src/ferc_xbrl_extractor/datapackage.py

```diff
@@ -336,7 +337,11 @@ def __init__(self, schema: Schema, period_type: str):
         self.columns = {
             field.name: CONVERT_DTYPES[field.type_] for field in schema.fields
         }
-        self.axes = [name for name in schema.primary_key if name.endswith("axis")]
+        self.axes = {
```

Comment:
We had been talking about trying to put the nullability data in the schema, but not wanting to make `fields: list[tuple[Field, bool]]` since that goes against the Frictionless definition of a datapackage.

Anyways. I found that there's this [`required` constraint](https://specs.frictionlessdata.io/table-schema/#constraints) on a `Field` which we could implement, and that would get this nullability information into the right spot.

Separately, what's the relationship between our datapackage classes and the official frictionless-py implementation of their data structures? I see that they support the constraints on a Field already: https://github.com/frictionlessdata/frictionless-py/blob/main/frictionless/schema/field.py#L22